### PR TITLE
Use git plugin 5.0.0 and git client plugin 4.0.0

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Use git plugin 5.0.0 and git client plugin 4.0.0

Changelogs for releases:

* https://github.com/jenkinsci/git-client-plugin/releases/tag/git-client-4.0.0
* https://github.com/jenkinsci/git-plugin/releases/tag/git-5.0.0

Previously evaluated in:

* https://github.com/jenkinsci/bom/pull/1619
* https://github.com/jenkinsci/bom/pull/1625

Pins git plugin 4.14.3 for Jenkins 2.346.x and older because git plugin 5.0.0 requires Java 11 and Jenkins 2.361.4 or newer.

Pins git client plugin 3.13.1 for Jenkins 2.346.x and older because git client plugin 4.0.0 requires Java 11 and Jenkins 2.361.4 or newer.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by MarkEWaite** [comment]
@timja or @jglick is it OK that I merge my own pull request or is it better that I wait for another reviewer to merge it?

I've labeled this pull request so that it will generate a new BOM release.  Let me know if that is not desirable.

**Review by jglick — APPROVED** [review]

**Comment by timja** [comment]
> @timja or @jglick is it OK that I merge my own pull request or is it better that I wait for another reviewer to merge it?

Yes go for it use your judgement for when something needs review, dependency updates are absolutely fine.

**Pull request merged**
